acceptancetests/jujupy/k8s_provider/kubernetes_core.py

```diff
@@ -0,0 +1,332 @@
+  namespace: kube-system
+subjects:
+  - kind: ServiceAccount
+    name: hostpath-provisioner
+    namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: hostpath-provisioner
+  apiGroup: rbac.authorization.k8s.io
+---
+
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: hostpath-provisioner
+  namespace: kube-system
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["create", "get", "delete"]
+---
+
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: RoleBinding
+metadata:
+  name: hostpath-provisioner
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: hostpath-provisioner
+subjects:
+  - kind: ServiceAccount
+    name: hostpath-provisioner
+---
+
+# -- Create a daemon set for web requests and send them to the nginx-ingress-controller
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: hostpath-provisioner
+  namespace: kube-system
+spec:
+  revisionHistoryLimit: 3
+  template:
+    metadata:
+      labels:
+        app: hostpath-provisioner
+    spec:
+      serviceAccountName: hostpath-provisioner
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: hostpath-provisioner
+          image: mazdermind/hostpath-provisioner:latest
+          imagePullPolicy: "IfNotPresent"
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: PV_DIR
+              value: /mnt/kubernetes
+          volumeMounts:
+            - name: pv-volume
+              mountPath: /mnt/kubernetes
+      volumes:
+        - name: pv-volume
+          hostPath:
+            path: /mnt/kubernetes
+---
+
+# -- Create the standard storage class for running on-node hostpath storage
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  # namespace: kube-system
+  name: {class_name}
+  annotations:
+    storageclass.beta.kubernetes.io/is-default-class: "true"
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: EnsureExists
+provisioner: hostpath
+---
+"""
+
+
+@register_provider
+class KubernetesCore(Base):
+
+    name = K8sProviderType.K8S_CORE
+
+    def __init__(self, client, timeout=1800):
+        super().__init__(client, timeout)
+        self.default_storage_class_name = "juju-storageclass"
+
+    def _ensure_kube_dir(self):
+        # ensure kube credentials
+        self.client.juju('scp', ('kubernetes-master/0:config', self.kube_config_path))
+
+        # ensure kubectl by scp from master
+        self.client.juju('scp', ('kubernetes-master/0:/snap/kubectl/current/kubectl', self.kubectl_path))
+
+    def _ensure_cluster_stack(self):
+        self.__ensure_lxd_profile(
+            LXD_PROFILE,
+            self.client.model_name  # current model now is the default IAAS hosted model for hosting k8s cluster.
+        )
+        self.__deploy_stack_bundle(BUNDLE)
```

Comment:
deploy the kubernetes-core bundle from the store

Reply:
changed